src/pkg/repository/repository_unexported_test.go

```diff
@@ -0,0 +1,57 @@
+package repository
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/alcionai/corso/src/internal/kopia"
+	"github.com/alcionai/corso/src/internal/tester"
+)
+
+type RepositoryModelSuite struct {
+	suite.Suite
+}
+
+func TestRepositoryModelSuite(t *testing.T) {
+	if err := tester.RunOnAny(
+		tester.CorsoCITests,
+		tester.CorsoRepositoryTests,
+	); err != nil {
+		t.Skip(err)
+	}
+
+	suite.Run(t, new(RepositoryModelSuite))
+}
+
+// ensure all required env values are populated
+func (suite *RepositoryModelSuite) SetupSuite() {
+	_, err := tester.GetRequiredEnvSls(
+		tester.AWSStorageCredEnvs,
+		tester.M365AcctCredEnvs)
+	require.NoError(suite.T(), err)
+}
+
+func (suite *RepositoryModelSuite) TestWriteGetModel() {
+	ctx, flush := tester.NewContext()
+	defer flush()
+
+	var (
+		t        = suite.T()
+		s        = tester.NewPrefixedS3Storage(t)
+		kopiaRef = kopia.NewConn(s)
+	)
+
+	require.NoError(t, kopiaRef.Connect(ctx))
+	defer kopiaRef.Close(ctx)
+
+	ms, err := kopia.NewModelStore(kopiaRef)
+	require.NoError(t, err)
+	require.NoError(t, newRepoModel(ctx, ms, "fnords"))
```

Comment:
nit: need to close the model store as well
```suggestion
	require.NoError(t, newRepoModel(ctx, ms, "fnords"))
	
	defer ms.Close(ctx)
```

Reply:
good catch, thanks